src/controllers/home.js

```diff
@@ -1,75 +1,77 @@
+}
+
+function getRequirePromise(deps) {
+    return new Promise(function (resolve, reject) {
+        require(deps, resolve);
+    });
+}
+
+function getTabController(index) {
+    if (null == index) {
+        throw new Error('index cannot be null');
     }
 
-    function getDefaultTabIndex() {
-        return 0;
-    }
+    const depends = [];
 
-    function getRequirePromise(deps) {
-        return new Promise(function (resolve, reject) {
-            require(deps, resolve);
-        });
-    }
-
-    function getTabController(index) {
-        if (null == index) {
-            throw new Error('index cannot be null');
-        }
+    switch (index) {
+        case 0:
+            depends.push('controllers/hometab');
+            break;
 
-        var depends = [];
+        case 1:
+            depends.push('controllers/favorites');
+    }
 
-        switch (index) {
-            case 0:
-                depends.push('controllers/hometab');
-                break;
+    const instance = this;
+    return getRequirePromise(depends).then(function (controllerFactory) {
+        let controller = instance.tabControllers[index];
 
-            case 1:
-                depends.push('controllers/favorites');
+        if (!controller) {
+            controller = new controllerFactory(instance.view.querySelector(".tabContent[data-index='" + index + "']"), instance.params);
+            instance.tabControllers[index] = controller;
         }
 
-        var instance = this;
-        return getRequirePromise(depends).then(function (controllerFactory) {
-            var controller = instance.tabControllers[index];
-
-            if (!controller) {
-                controller = new controllerFactory(instance.view.querySelector(".tabContent[data-index='" + index + "']"), instance.params);
-                instance.tabControllers[index] = controller;
-            }
-
-            return controller;
-        });
-    }
+        return controller;
+    });
+}
 
-    function HomeView(view, params) {
+class HomeView {
+    constructor(view, params) {
         TabbedView.call(this, view, params);
     }
-
-    Object.assign(HomeView.prototype, TabbedView.prototype);
-    HomeView.prototype.getTabs = getTabs;
-    HomeView.prototype.getDefaultTabIndex = getDefaultTabIndex;
-    HomeView.prototype.getTabController = getTabController;
-
-    HomeView.prototype.setTitle = function () {
+    setTitle() {
         Emby.Page.setTitle(null);
-    };
-
-    HomeView.prototype.onPause = function () {
+    }
+    onPause() {
         TabbedView.prototype.onPause.call(this);
         document.querySelector('.skinHeader').classList.remove('noHomeButtonHeader');
-    };
-
-    HomeView.prototype.onResume = function (options) {
+    }
+    onResume(options) {
         TabbedView.prototype.onResume.call(this, options);
         document.querySelector('.skinHeader').classList.add('noHomeButtonHeader');
-    };
+    }
+}
+
+Object.assign(HomeView.prototype, TabbedView.prototype);
```

Comment:
With the extend, this and all the `TabbedView.prototype.whatever.call` aren't needed.